src/TritonRoute/src/pa/FlexPA_prep.cpp

```diff
@@ -854,49 +855,42 @@ void FlexPA::prepPoint_pin_checkPoint_via(
     auto via = make_unique<frVia>(viaDef);
     via->setOrigin(bp);
     via->getLayer1BBox(box);
+    if (instTerm) {
+        if (!boundaryBBox.contains(box))
+            continue;
+        frBox layer2BBox;
+        via->getLayer2BBox(layer2BBox);
+        if (!boundaryBBox.contains(layer2BBox))
```

Comment:
Why are we checking both layers are enclosed without checking if they are in the viainpin range?

Reply:
This check is against the cell bbox. The check already existed, but it was done after the viaInpin check.